src/Deinterleave.cpp

```diff
@@ -228,7 +245,27 @@ class Deinterleaver : public IRGraphMutator {
     }
 
     Expr visit(const Ramp *op) override {
-        Expr expr = op->base + starting_lane * op->stride;
+        int base_lanes = op->base.type().lanes();
+        if (base_lanes > 1) {
+            if (new_lanes == 1) {
+                int index = starting_lane / base_lanes;
+                Expr expr = op->base + cast(op->base.type(), index) * op->stride;
+
+                int old_starting_lane = starting_lane;
```

Comment:
using ScopedValue could be cleaner here

Reply:
Updated, it indeed looks much cleaner now, thank you for suggestion!